qt/python/mantidqt/mantidqt/plotting/functions.py

```diff
@@ -342,7 +345,11 @@ def plot_surface(workspaces, fig=None):
             fig.clf()
             ax = fig.add_subplot(111, projection='mantid3d')
         else:
-            fig, ax = plt.subplots(subplot_kw={'projection': 'mantid3d'})
+            if LooseVersion('3.4.0') <= LooseVersion(matplotlib.__version__):
+                fig, ax = plt.subplots(subplot_kw={'projection': 'mantid3d', "auto_add_to_figure": False})
```

Comment:
On the surface this seems strange to say `auto_add_to_figure` is `False` and then add it? Could you help me understand why? If necessary could you add a comment?

Reply:
As of 3.4.0 the behaviour of automatically adding to figure for Axis3D has been deprecated and prints a warning explaining such, then suggesting to do this. This way the behaviour is maintained that we had prior, whilst hiding the deprecation warning, and in future (matplotlib 3.6+) we will probably need to remove `auto_add_to_figure`.